src/app/components/collections/collections.component.ts

```diff
@@ -31,14 +32,18 @@ export class CollectionsComponent implements OnInit, OnDestroy {
       }
     });
 
-    this.bookshelf$.pipe(take(2)).subscribe((bookshelf) => {
-      if (bookshelf && bookshelf.bookshelfId !== '') {
-        this.store.dispatch(getBooks({ bookshelfId: bookshelf.bookshelfId }));
-      }
-    });
+    this.bookshelfSubscription = this.bookshelf$
+      .pipe(take(2))
```

Comment:
you both unsubscribe and take(2) here, both which complete the observable. Only one is necessary, it depends on the lifespan you want the observable to have.

Reply:
Thank you, I think I was testing something here. I removed the subscription.